src/MICore/MIResults.cs

```diff
@@ -730,12 +730,20 @@ private Results ParseResultList(Span listStr, ResultClass resultClass = ResultCl
                     {
                         return i == s.Extent;
                     }, listStr, out rest);
+
+            Results results = new Results(resultClass, list);
+
             if (!rest.IsEmpty)
```

Comment:
I would change this if statement so it isn't a negation since you are now doing more work here.

```
if (rest.IsEmpty)
{ 
    return results;
}

// your other stuff now
```